homeassistant/components/wilight/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow to configure WiLight."""
+import logging
+from urllib.parse import urlparse
+
+import pywilight
+
+from homeassistant.components import ssdp
+from homeassistant.config_entries import CONN_CLASS_LOCAL_PUSH, ConfigFlow
+from homeassistant.const import CONF_HOST
+
+from . import PLATFORMS
+from .const import DOMAIN  # pylint: disable=unused-import
+
+CONF_SERIAL_NUMBER = "serial_number"
+CONF_MODEL_NAME = "model_name"
+
+WILIGHT_MANUFACTURER = "All Automacao Ltda"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WiLightFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a WiLight config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_PUSH
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+    def __init__(self):
+        """Initialize the WiLight flow."""
+        self._host = None
+        self._serial_number = None
+        self._device_id = None
+        self._model_name = None
+        self._components = []
+        self._components_text = ""
+
+    def _self_update(self, host, serial_number, model_name):
+        self._host = host
+        self._serial_number = serial_number
+        self._device_id = f"WL{serial_number}"
```

Comment:
We don't need to add `WL` to the unique_id. The core is aware of the integration domain when comparing unique ids.

Reply:
I prefer to keep the WL, because that's how it appears in the integrations.
If it is not an impediment, I would like to maintain.

Reply:
Use a separate title not the same as the unique_id for the entry.

Reply:
I'm thinking of replacing self._device_id with self._tittle and on line 85: await self.async_set_unique_id (self._device_id)
change to:
await self.async_set_unique_id (self._serial_number)

What do you think?

Reply:
Done